File: 6. Zigzag Conversion.java
class Solution {
    public String convert(String s, int numRows) {

        if(numRows==1)
        {
            return s;
        }

        StringBuilder builder = new StringBuilder();
        int len = s.length();
        int section1 = 2* (numRows -1);

        for(int i=0;i<numRows;i++)
        {
            int index = i ;
            while(index<len)
            {
                builder.append(s.charAt(index));
                if(i!=0 && i!=numRows-1)
                {
                    int section2 = section1- (2*i);
                    int section3 = index + section2;

                    if(section3<len){
                        builder.append(s.charAt(section3));
                    }
                }

                index = index + section1;
                
            }
        }
        return builder.toString();
        
    }
}

File: python_codes/6. Zigzag Conversion.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        if numRows >= len(s) or len(s)<=1 or numRows==1:
            return s
        
        rows = [""]*numRows
        row = 0
        step = 1

        for c in s:
            rows[row] += c
            if row == 0:
                step = 1
            elif row == numRows - 1:
                step = -1
            row +=step

        return "".join(rows)